hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+import matplotlib.cm as cm
+from hyperspy._signals.signal2d import Signal2D
+
+
+class HologramImage(Signal2D):
+    """Image subclass for holograms acquired via electron holography."""
+
+    _signal_type = 'hologram'
+
+    def reconstruct_phase(self, reference=None, rec_param=None, show_phase=False, return_param=False,
+                          fresnel_fringe_filter=False, verbose=False):
+        """Reconstruct holography data
+
+        Parameters
+        ----------
+        reference : ndarray, class:`~hyperspy._signals.Image`, `~hyperspy._signals.HologramImage`
+            Vacuum reference hologram.
+        rec_param : tuple
+            Reconstruction parameters in sequence (SBrect(x0, y0, x1, y1), SB size)
+        show_phase : boolean
+            set True to plot phase after the reconstruction
+        return_param : boolean
+            set True to return reconstruction as tuple, useful for batch processing
+        fresnel_fringe_filter : boolean
+            set True to apply an automatic filtering of biprisms fresnel fringe
+        verbose : boolean
+            set True to see details of the reconstruction (i.e. SB selection)
+
+        Returns
+        -------
+        wave : WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+        rec_param : tuple
+            see the description in Parameters (optional, triggered by return_param flag)
+
+        Notes
+        -----
+        The reconstruction parameters assigned interactively if rec_param is not givven
+
+        See Also
+        --------
+        _reconstruct
+
+        """
+
+        (sx, sy) = self.data.shape
+        holo_data = self.data
+
+        # Parsing reference input:
+        if reference is None:  # to reconstruct hologram WITHOUT a reference
+            eh_hw_fft = fftshift(fft2(holo_data))
+        elif isinstance(reference, Signal2D):
+            # to reconstruct hologram WITH a reference
+            ref_data = reference.data
+            eh_hw_fft = fftshift((fft2(ref_data)))
+        else:
+            ref_data = reference
+            eh_hw_fft = fftshift((fft2(ref_data)))
+
+        # Preparing reconstruction parameters:
+        if rec_param is None:
+            eh_hw_fft_half = eh_hw_fft[0:np.int(sx / 2.02), :]  # selects half of the fft excluding fraction of points
+            #  in the middle
+            yr, xc = np.unravel_index(eh_hw_fft_half.argmax(), eh_hw_fft_half.shape)  # Center of selected sideBand
+            sb_pos = [yr, xc]
+            # Default SB size is 1/2 of the distance to main band
+            sb_size = np.round(1.0 / 2 * (norm(np.subtract(sb_pos, [sx / 2, sy / 2]))))
+            # To be sure of even number, still questionable if it is needed:
+            sb_size = sb_size - np.mod(sb_size, 2)
+            rec_param = (xc - 1, yr - 1, xc + 1, yr + 1, sb_size)
+            print("Sideband Size in pixels")
+            print("%d" % sb_size)
```

Comment:
You said that you want `sb_size` to be an integer and I think here is the place where you should make sure of that using the following alternative:
```python
sb_size = norm(np.subtract(sb_pos, [sx / 2, sy / 2])) // 4 * 2
```

The `// 4 * 2` part will make sure of three things all at once:
- `sb_size` is halve of the distance to the main band.
- `sb_size` is an integers (`//` is integer division).
- `sb_size` is an even numbers so you don't need `sb_size = sb_size - np.mod(sb_size, 2)`.


Also I assume loggers instead of print statements would be better in this case.
```python
import logging                         # These first two lines
_logger = logging.getLogger(__name__)  # go to the import section

_logger.info('Sideband size in pixels: {}'.format(sb_size))
```
------
Also, why is the SB rectangle you set with the first four entries of `rec_param` only 3 pixels big? Is it because of the automated peak finding instead of setting the rectangle via a GUI?